### DISCUSSION TITLE
[Kavita]  Support for Plain Text Files (*.txt)

### DISCUSSION THREAD

**Discussion by TenviLi** [Kareadita/Kavita#3201] [Ideas]
#### Description:

I have a large collection of novels stored in plain text format (*.txt). Currently, I am unable to manage these novels with Kavita unless I convert them to *.epub format, which can be time-consuming.

It would be extremely helpful if Kavita could natively support plain text files (*.txt) as a readable format. This feature would allow me and others with similar collections to manage and enjoy our novels without the need for conversion.

#### Benefits:

- Simplifies the management of text-based novels without requiring conversion.
- Saves time for users who have extensive libraries in *.txt format.
- Expands Kavita’s versatility in handling different text file formats.

Thank you for considering this enhancement!

---

```
Extensive File support
Manga/Comics: ZIP, RAR, RAR5, CBR, CBZ, CB7, CBT, TAR.GZ, 7ZIP/7Z
Books: EPUB2, EPUB3, PDF
Raw Images: JPG/JPEG, PNG, WebP, GIF, AVIF
```

### Idea Category

User Experience

### Duration of Using Kavita

3 years

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by smith113-p** [comment]
note that there is a duplicate request for this, #2561

**Comment by majora2007** [comment]
I'm not saying we will implement this, but I was shown a personal fork where this was done. It can be done by creating an epub on the fly for a text file, each newline becomes a separate paragraph to allow progress tracking. Metadata support would be hard, since there is a strict no sidecar policy in Kavita.

**Comment by smith113-p** [comment]
Speaking for myself only:

Converting to epub sounds good to me. Although I would prefer that only empty lines separate paragraphs, e.g.:

```
This is some text
split across multiple
lines.

Now here is a new
paragraph
```

For metadata, using the filename as the title or maybe allowing a "Author - Title.txt" convention would work for me.

**Reply by scare376** [reply]
Why would it be more ideal? What if the writer doesn't use an empty line to distinguish paragraphs. Then the entire text file would be a single massive paragraph.

**Reply by scare376** [reply]
Newline means where the writer would hit enter. Word wrapping is dependant on resolution at that point to determine lines

**Reply by smith113-p** [reply]
In the files that I have/have seen, it is more common that the author uses blank lines to separate paragraphs, but maybe that is a minority, idk
